frontends/election-manager/src/screens/print_test_deck_screen.test.tsx

```diff
@@ -1,32 +1,104 @@
+  PrintTestDeckScreen,
+  TWO_SIDED_PAGE_PRINT_TIME_MS,
+} from './print_test_deck_screen';
 import { renderInAppContext } from '../../test/render_in_app_context';
 
 jest.mock('../components/hand_marked_paper_ballot');
 
-beforeAll(() => {
-  window.kiosk = fakeKiosk();
+let mockKiosk: jest.Mocked<KioskBrowser.Kiosk>;
+let mockLogger: Logger;
+let mockPrintBallotRef: RefObject<HTMLElement>;
+let mockPrinter: jest.Mocked<Printer>;
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  mockKiosk = fakeKiosk();
+  mockKiosk.getPrinterInfo.mockResolvedValue([
+    fakePrinterInfo({ connected: true, name: 'VxPrinter' }),
+  ]);
+  window.kiosk = mockKiosk;
+  mockLogger = new Logger(LogSource.VxAdminFrontend, mockKiosk);
+  mockPrintBallotRef = {
+    current: document.createElement('div'),
+  };
+  mockPrinter = fakePrinter();
 });
 
 afterAll(() => {
   delete window.kiosk;
 });
 
-test('Printing all L&A packages sorts precincts', async () => {
-  jest.useFakeTimers();
-  const mockKiosk = window.kiosk! as jest.Mocked<KioskBrowser.Kiosk>;
-  mockKiosk.getPrinterInfo.mockResolvedValue([
-    fakePrinterInfo({ name: 'VxPrinter', connected: true }),
-  ]);
+test('Printing L&A package for one precinct', async () => {
+  renderInAppContext(<PrintTestDeckScreen />, {
+    logger: mockLogger,
+    printBallotRef: mockPrintBallotRef,
+    printer: mockPrinter,
+  });
 
+  userEvent.click(screen.getByText('District 5'));
+
+  const printText = 'Printing L&A Package: District 5';
+  await screen.findByText(printText);
+  await waitFor(() => expect(mockPrinter.print).toHaveBeenCalledTimes(1));
+  expect(mockPrinter.print).toHaveBeenLastCalledWith({ sides: 'one-sided' });
+  await waitFor(() =>
+    expect(mockKiosk.log).toHaveBeenLastCalledWith(
+      expect.stringContaining(LogEventId.TestDeckTallyReportPrinted)
+    )
+  );
+  jest.advanceTimersByTime(ONE_SIDED_PAGE_PRINT_TIME_MS + 1);
+
+  await screen.findByText(printText);
+  await waitFor(() => expect(mockPrinter.print).toHaveBeenCalledTimes(2));
+  expect(mockPrinter.print).toHaveBeenLastCalledWith({ sides: 'one-sided' });
+  await waitFor(() =>
+    expect(mockKiosk.log).toHaveBeenLastCalledWith(
+      expect.stringContaining(LogEventId.TestDeckPrinted)
+    )
+  );
+  expect(mockKiosk.log).toHaveBeenLastCalledWith(
+    expect.stringContaining('BMD paper ballot test deck printed')
+  );
+  jest.advanceTimersByTime(3 * ONE_SIDED_PAGE_PRINT_TIME_MS + 1);
+
+  await screen.findByText(printText);
+  await waitFor(() => expect(mockPrinter.print).toHaveBeenCalledTimes(3));
+  expect(mockPrinter.print).toHaveBeenLastCalledWith({
+    sides: 'two-sided-long-edge',
+  });
+  await waitFor(() =>
+    expect(mockKiosk.log).toHaveBeenLastCalledWith(
+      expect.stringContaining(LogEventId.TestDeckPrinted)
+    )
+  );
+  expect(mockKiosk.log).toHaveBeenLastCalledWith(
+    expect.stringContaining('Hand-marked paper ballot test deck printed')
+  );
+  jest.advanceTimersByTime(LAST_PRINT_JOB_SLEEP_MS + 1);
```

Comment:
Are these `+ 1`s actually needed?

Reply:
Good q - I was following the example of other tests:

https://github.com/votingworks/vxsuite/blob/0626f6ce583ecb1933e627ea19c9ca65f6f5d197/frontends/election-manager/src/app.test.tsx#L445

But will do some testing to verify whether these 1ms additions are really necessary (gut tells me they shouldn't be)

Reply:
Yeah I don't think they should be, those examples are usually my fault when I get frustrated by things not working and start randomly adding/increasing jest.advanceTimesByTime lol